engine/src/test/scala/cromwell/engine/workflow/lifecycle/WorkflowExecutionActorSpec.scala

```diff
@@ -0,0 +1,52 @@
+package cromwell.engine.workflow.lifecycle
+
+import akka.testkit.EventFilter
+import com.typesafe.config.ConfigFactory
+import cromwell.CromwellTestkitSpec
+import cromwell.core.WorkflowId
+import cromwell.engine.backend.{BackendConfigurationEntry, CromwellBackend, WorkflowDescriptorBuilder}
+import cromwell.engine.workflow.lifecycle.WorkflowExecutionActor.StartExecutingWorkflowCommand
+import cromwell.util.SampleWdl
+import org.scalatest.BeforeAndAfter
+
+class WorkflowExecutionActorSpec extends CromwellTestkitSpec with BeforeAndAfter with WorkflowDescriptorBuilder {
+
+  override implicit val actorSystem = system
+
+  val stubbedConfig = ConfigFactory.load().getConfig("backend.providers.Mock")
+
+  val MockBackendConfigEntry = BackendConfigurationEntry(
+    name = "Mock",
+    className = stubbedConfig.getString("class"),
+    shadowLifecycleActorFactoryClass = stubbedConfig.getString("actor-factory"),
+    stubbedConfig.getConfig("config")
+  )
+
+  CromwellBackend.initBackends(List(MockBackendConfigEntry), MockBackendConfigEntry, system, shadowExecutionEnabled = true)
+
+
+  val runtimeSection =
+    """
+      |runtime {
+      | backend: "Mock"
+      |}
+    """.stripMargin
+
+  val workflowId = WorkflowId.randomId()
+  val engineWorkflowDescriptor = createMaterializedEngineWorkflowDescriptor(workflowId, SampleWdl.HelloWorld.asWorkflowSources(runtime = runtimeSection))
+
+  "WorkflowExecutionActor" should {
```

Comment:
I wanted to write more unit tests but it's gotten very difficult to write more without repeating lots of code (like creating new Mock backends). The `CromwellBackend` singleton is used in the WorkflowExecutionActor (and a few other places) which prevent's its (stubbed-out) injection into these classes.